react/react1/week2/Todo/my-app/src/App.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from "react";
+import data from "./data.json";
+
+//components
+import Header from "./Header";
+import ToDoList from "./ToDoList";
+import ToDoForm from "./ToDoForm";
+import Timer from "./Timer";
+
+import "./App.css";
+
+function App() {
+  const [toDoList, setToDoList] = useState(data);
+
+  const handleToggle = (id) => {
+    let mapped = toDoList.map((task) => {
+      return task.id === Number(id)
+        ? { ...task, complete: !task.complete }
+        : { ...task };
+    });
+    setToDoList(mapped);
+  };
+
+  const handleFilter = () => {
+    let filtered = toDoList.filter((task) => {
+      return !task.complete;
+    });
+    setToDoList(filtered);
+  };
+
+  const addTask = (userInput) => {
+    let copy = [...toDoList];
+    copy = [
+      ...copy,
+      { id: toDoList.length + 1, task: userInput, complete: false },
```

Comment:
Defining id this way will make your app stuck or crash.
For ex, if you have a list of length 5 with is of the last item be 4(according to your formula), you deleted an item somewhere in between for ex item with id 2.
When you try adding a new item your app tries to create an id 4 again(according to your formula), but we already have an item with the same id, that is when the app gets errors.
There are a lot of other methods to create a unique id. Google it and try another one.